sbndcode/OpDetSim/DigiPMTSBNDAlg.cc

```diff
@@ -141,7 +145,10 @@ namespace opdet {
         tphoton = ttsTime + simphotons[i].Time - t_min + ttpb + fParams.CableTime;
         if(tphoton < 0.) continue; // discard if it didn't made it to the acquisition
         timeBin = std::floor(tphoton*fSampling);
-        if(timeBin < wave.size()) {AddSPE(timeBin, wave);}
+        if(timeBin < wave.size()){
+          if(fParams.MakeGainFluctuations) AddSPE(timeBin, wave, fPMTGainFluctuationsPtr->GainFluctuation(1,fEngine));
+          else AddSPE(timeBin, wave);
+        }
```

Comment:
These two (or four) lines are very popular:
```suggestion
        AddSPEwithFluctuations(wave, timeBin);
```
They appear in the code four times. I **recommend** that they be put into a member function.